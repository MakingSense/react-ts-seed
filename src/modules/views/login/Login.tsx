import React, { memo, useCallback, useState } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface ILoginProps {
  login: (username: string, password: string) => void;
}

export interface ILoginState {
  username: string;
  password: string;
}

const Login = ({ login }: ILoginProps) => {
  const [state, setState] = useState<ILoginState>({ username: '', password: '' });

  const onLogin = useCallback(() => login(state.username, state.password), [state, login]);

  const setUsername = useCallback((event: React.ChangeEvent<HTMLInputElement>) => setState({ ...state, username: event.target.value }), [state, setState]);

  const setPassword = useCallback((event: React.ChangeEvent<HTMLInputElement>) => setState({ ...state, password: event.target.value }), [state, setState]);

  return (
    <div className={`${css(styles.loginContainer)} container`} key="Login">
      <form className={`${css(styles.loginForm)} form-group`}>
        <input
          className={`${css(styles.loginInput)} form-control`}
          data-testid="login-username"
          onChange={setUsername}
          value={state.username}
          placeholder="username"
          type="text"
          autoComplete="email"
        />
        <input
          className={`${css(styles.loginInput)} form-control`}
          onChange={setPassword}
          value={state.password}
          data-testid="login-password"
          placeholder="password"
          type="password"
          autoComplete="password"
        />
        <button data-testid="login-btn" className={`${css(styles.loginButton)} btn btn-primary`} onClick={onLogin} type="button">
          Login
        </button>
      </form>
    </div>
  );
};

export default memo(Login);
