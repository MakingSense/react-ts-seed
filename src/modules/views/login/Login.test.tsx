import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import Login from './Login';

describe('Login Component', () => {
  let Component;
  let props;

  beforeEach(() => {
    global.console.error = () => {/** */};
    props = {
      login: jest.fn()
    };
    Component = mount(<Login {...props} />);
  });

  it('should render with default props', () => {
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should set username and password and login', () => {
    const username = 'username';
    const password = 'password';
    Component.find('input[type="text"]').simulate('change', { target: { value: username } });
    Component.find('input[type="password"]').simulate('change', { target: { value: password } });
    Component.find('button').simulate('click');
    expect(props.login).toBeCalledWith(username, password);
  });
});
