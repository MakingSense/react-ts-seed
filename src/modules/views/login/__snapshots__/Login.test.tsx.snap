// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render with default props 1`] = `
.loginContainer {
  -webkit-box-align: center;
  -ms-flex-align: center;
  height: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
}

.loginForm {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  align-items: center;
}

.loginInput {
  margin-bottom: 1rem;
}

.loginButton {
  width: 50%;
}

<div
  className="loginContainer container"
>
  <form
    className="loginForm form-group"
  >
    <input
      autoComplete="email"
      className="loginInput form-control"
      data-testid="login-username"
      onChange={[Function]}
      placeholder="username"
      type="text"
      value=""
    />
    <input
      autoComplete="password"
      className="loginInput form-control"
      data-testid="login-password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <button
      className="loginButton btn btn-primary"
      data-testid="login-btn"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </form>
</div>
`;
